src/Database/LSMTree/Internal.hs

```diff
@@ -907,6 +1011,7 @@ addRunToLevels conf@TableConfig{..} resolve hfs hbio root uniqC r0 reg levels =
           LevelTiering -> do
             let mergelast = mergeLastForLevel ls
             mr' <- newMerge LevelTiering mergelast ln rs'
+            traceWith tr $ TraceAddRun ln (Run.runRunFsPaths r)
```

Comment:
> This sounds like `r` is being added to the level, but it's just being moved from `mr` to the other runs. `rs'` are being added.

Information about `rs'` is already represented in the `TraceNewMerge` trace. Conceptually, moving `r` to the other runs is adding the run to this level: AFAIU we represent incoming merges on the next level, but they technically are still part of the previous level. So if we view it like that, we should be tracing new merges at level `ln - 1` instead of `ln` like we are doing now.

> I also think it's useful to see how many runs are in the level already, or even how big they are. So we could include `1 + V.length rs` or `map runNumEntries (r:rs)` into the trace as well.

Alright, I'm tracing the run numbers of the resident runs now as well

> And maybe we should trace this whenever runs are pushed to the next level, so as the first thing in `go`. Then we always see how runs are being moved and sometimes this will additionally result in a merge being created.

TODO